### PR TITLE
refs https://github.com/metasfresh/metasfresh/issues/14165 adjust woS…

### PR DESCRIPTION
**Pull request by fp-ps**
…tep dateStart and dateEnd

https://github.com/metasfresh/metasfresh/issues/14165

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java [hunk 1]
```diff
@@ -2230,4 +2230,40 @@ public static long getMillisBetween(@NonNull final Timestamp timestamp1, @NonNul
 	{
 		return timestamp2.getTime() - timestamp1.getTime();
 	}
+
+	@Nullable
+	public static Instant asStartOfDayInstant(@Nullable final LocalDate localDate, @NonNull final ZoneId zoneId)
+	{
+		if(localDate == null)
```
Comment (cp-ps) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java:2237]: pls format the code in the whole file

**Review by cp-ps — APPROVED** [review]

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java [hunk 2]
```diff
@@ -2230,4 +2230,40 @@ public static long getMillisBetween(@NonNull final Timestamp timestamp1, @NonNul
...
 	}
+
+	@Nullable
+	public static Instant asStartOfDayInstant(@Nullable final LocalDate localDate, @NonNull final ZoneId zoneId)
+	{
+		if (localDate == null)
+		{
+			return null;
+		}
+		final LocalDateTime endOfDay = localDate.atTime(LocalTime.MIN);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java:2241]: @fp-ps shouldn't be LocalTime.MAX?

Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java:2241]: ahhh.... the `endOfDay` variable shall be renamed, i guess :)

Comment (fp-ps) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java]: yees, I renamed it

**Review by metas-ts — APPROVED** [review]

**Pull request merged**